File: BOJ/yein/2_week/덩치.py
N = int(input())
arr = []
for i in range(N):
    arr.append(list(map(int, input().split())))

rank = [1 for _ in range(N) ]

for i in range(N):
    for j in range(N):
        if arr[i][0] < arr[j][0] and arr[i][1] < arr[j][1]:
            rank[i] += 1

print(*rank)

File: BOJ/jeongbeen/2_week/덩치.java
import java.io.*;
import java.util.*;

public class 덩치 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int[][] arr = new int[N][2];
        Map<List<Integer>, Integer> map = new HashMap<>();

        for (int i = 0; i < N; i++) {
            String[] s = br.readLine().split(" ");
            arr[i][0] = Integer.parseInt(s[0]);
            arr[i][1] = Integer.parseInt(s[1]);
        }

        for (int i = 0; i < N; i++) {
            int rank = 1;
            for (int j = 0; j < N; j++) {
                if (arr[i][0] < arr[j][0] && arr[i][1] < arr[j][1]) {
                    rank++;
                }
            }
            bw.write(rank + " ");
        }

        bw.flush();
        br.close();
        bw.close();
    }
}
